src/lib.rs

```diff
@@ -126,15 +126,49 @@ pub trait Renderable: Send + Sync {
     fn render(&self, context: &mut Context) -> Result<Option<String>>;
 }
 
+pub type TemplateLocation = Path;
+pub trait FileRepository {
```

Comment:
I feel like calling it a `FileRepository` is leaking the implementation details.  Maybe `TemplateRepository`?

Now for some more specific parts of the name

`Template`: I think this will be fine.  Upon reviewing this, I looked up what liquid refers to what `include` pulls in.  They don't formally name their parameters for the `include` tag that I saw but referred to it as a snippet.  Their source code refers to it as a template.  Either way, I have bigger ambitions for this.  I think it'd be useful for caching at which point it will be reused both for parsing top level files and for parsing `include`d files.  At that point, `Template` works.

`Repository`: I think this will be fine.  My gut tells me that there are some common terms for this type of thing but I can't think of them atm.  Maybe `Repository` is one.  Either way, if we don't have another idea for a name, I think this `Repository` will be fine.